lib/src/main/java/org/openbase/bco/dal/lib/layer/unit/Unit.java

```diff
@@ -741,4 +712,7 @@ default ServiceProvider getServiceProvider() {
     default List<ActionDescription> getActionList() throws NotAvailableException {
         return ProtoBufFieldProcessor.getRepeatedFieldList(Action.TYPE_FIELD_NAME_ACTION, getData());
     }
+
+    @RPCMethod
+    Future<AggregatedServiceStateType.AggregatedServiceState> queryAggregatedServiceState(final DatabaseQueryType.DatabaseQuery databaseQuery);
```

Comment:
@pLeminoq could you please assist @DavidWein94 to encrypt this the `AggregatedServiceState` and the `DatabaseQuery` by using the bco.authentication module? Furthermore, the read permission of the unit need to be validated on controller side. Otherwise, this method would allow everyone to request any unit states independent of the configured unit permissions.

Reply:
issue created, therefore issue will not be a part of this pr.